shenyu-integrated-test/shenyu-integrated-test-combination/src/test/java/org/apache/shenyu/integrated/test/combination/SharedThreadPoolTest.java

```diff
@@ -0,0 +1,126 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * The integrated test for combination plugins about shared thread pool.
+ */
+public class SharedThreadPoolTest extends AbstractPluginDataInit {
+    
+    private static final Logger LOG = LoggerFactory.getLogger(SharedThreadPoolTest.class);
+    
+    @BeforeAll
+    public static void setup() throws IOException {
+        // for apache dubbo
+        String pluginResult = initPlugin(PluginEnum.DUBBO.getName(), "{\"register\":\"zookeeper://shenyu-zk:2181\",\"threadpool\": \"shared\"}");
+        assertThat(pluginResult, is("success"));
+        // for grpc
+        pluginResult = initPlugin(PluginEnum.GRPC.getName(), "{\"register\":\"zookeeper://shenyu-zk:2181\",\"threadpool\": \"shared\"}");
+        assertThat(pluginResult, is("success"));
+        // for motan
+        pluginResult = initPlugin(PluginEnum.MOTAN.getName(), "{\"register\":\"shenyu-zk:2181\",\"threadpool\": \"shared\"}");
+        assertThat(pluginResult, is("success"));
+        // for sofa
+        pluginResult = initPlugin(PluginEnum.SOFA.getName(), "{\"protocol\":\"zookeeper\",\"register\":\"shenyu-zk:2181\",\"threadpool\": \"shared\"}");
+        assertThat(pluginResult, Matchers.is("success"));
+    }
+    
+    @Test
+    public void testApacheDubbo() throws IOException {
+        DubboTest dubboTest = HttpHelper.INSTANCE.getFromGateway("/dubbo/findById?id=1", DubboTest.class);
+        assertEquals("hello world shenyu Apache, findById", dubboTest.getName());
+        assertEquals("1", dubboTest.getId());
+    }
+    
+    @Test
+    public void testGrpc() throws Exception {
+        JsonArray jsonArray = new JsonArray();
+        JsonObject child = new JsonObject();
+        child.addProperty("message", "hello rpc");
+        jsonArray.add(child);
+        JsonObject request = new JsonObject();
+        request.add("data", jsonArray);
+        JsonArray response = HttpHelper.INSTANCE.postGateway("/grpc/echo", request, JsonArray.class);
+        Map<String, Object> result = GsonUtils.getInstance().toObjectMap(response.get(0).toString(), Object.class);
+        assertEquals("ReceivedHELLO", result.get("message"));
+    }
+    
+    @Test
+    public void testMotan() throws Exception {
+        MotanDTO request = new MotanDTO("shenyu");
+        Type returnType = new TypeToken<String>() {
+        }.getType();
+        String response = HttpHelper.INSTANCE.postGateway("/motan/demo/hello", request, returnType);
+        assertEquals("hello shenyu", response);
+    }
+    
+    @Test
+    public void testSofa() {
+        try {
+            SofaTestData response = HttpHelper.INSTANCE.getFromGateway("/sofa/findById?id=1001", new TypeToken<SofaTestData>() {
```

Comment:
There is no problem in other places, that is, the catch is used here, and I see that the sofa test in the log will make an error. I suggest to solve the problem of the plug-in first and then merge the pr.

Reply:
em...I tried at best, exclude dependencies, upgrade sofa version, all not work. How about remove `sofa test part` from `SharedThreadPoolTest`? I think `#testApacheDubbo`, `#testGrpc`, `#testMotan` and `#testIsOneThreadPool` has proved enugth.

Reply:
Maybe, we can add a TODO comment here.